bridge/svix-bridge-plugin-kafka/src/input.rs

```diff
@@ -0,0 +1,215 @@
+        let payload = msg.payload().ok_or_else(|| Error::MissingPayload)?;
+        let payload = if let Some(transformation) = &self.transformation {
+            let input = match transformation.format() {
+                TransformerInputFormat::Json => {
+                    let json_payload =
+                        serde_json::from_slice(payload).map_err(Error::Deserialization)?;
+                    TransformerInput::JSON(json_payload)
+                }
+                TransformerInputFormat::String => {
+                    let raw_payload = str::from_utf8(payload).map_err(Error::NonUtf8Payload)?;
+                    TransformerInput::String(raw_payload.to_string())
+                }
+            };
+
+            let script = transformation.source().clone();
+            let object = self.transform(script, input).await?;
+            serde_json::from_value(serde_json::Value::Object(object))
+                .map_err(Error::Deserialization)?
+        } else {
+            serde_json::from_slice(payload).map_err(Error::Deserialization)?
+        };
+
+        let CreateMessageRequest {
+            app_id,
+            message,
+            post_options,
+        } = payload;
+
+        // FIXME(jplatte): I think we should always set the idempotency key
+
+        self.svix_client
+            .message()
+            .create(app_id, message, post_options.map(Into::into))
+            .await?;
+
+        Ok(())
+    }
+
+    async fn transform(&self, script: String, input: TransformerInput) -> Result<JsObject> {
+        let (job, rx) = TransformerJob::new(script, input);
+        self.transformer_tx
+            .as_ref()
+            .ok_or_else(|| Error::transformation("transformations not configured"))?
+            .send(job)
+            .map_err(|e| Error::transformation(e.to_string()))?;
+
+        let ret = rx
+            .await
+            .map_err(|_e| Error::transformation("transformation rx failed"))
+            .and_then(|x| {
+                x.map_err(|_e| Error::transformation("transformation execution failed"))
+            })?;
+
+        match ret {
+            TransformerOutput::Object(v) => Ok(v),
+            TransformerOutput::Invalid => Err(Error::transformation(
+                "transformation produced unexpected value",
+            )),
+        }
+    }
+
+    async fn run_inner(&self) -> Result<()> {
+        let opts = self.opts.clone();
+        // `ClientConfig::create` does blocking I/O.
+        // Same for subscribe, most likely.
+        let consumer = spawn_blocking(move || {
+            let topic = opts.topic.clone();
+
+            let consumer = opts.create_consumer()?;
+            tracing::debug!("Created StreamConsumer");
+
+            consumer.subscribe(&[&topic])?;
+            tracing::debug!(topic, "Subscribed");
+
+            Ok::<_, KafkaError>(consumer)
+        })
+        .await
+        // unwrap propagates any panics from create_consumer caught by tokio
```

Comment:
```suggestion
```

I think instead of a comment, it would be simpler to just embed the rationale in the panic message. e.g.

`.expect("create_consumer task panicked")`

Or something similar